docs/source/cpp/parquet.rst

```diff
@@ -27,11 +27,25 @@ Reading and writing Parquet files
 The Parquet C++ library is part of the Apache Arrow project and benefits
 from tight integration with Arrow C++.
 
-Reading
-=======
+The :class:`arrow::FileReader` class reads data for an entire
+file or row group into an :class:`::arrow::Table`.
 
-The Parquet :class:`FileReader` requires a :class:`::arrow::io::RandomAccessFile`
-instance representing the input file.
+The :func:`arrow::WriteTable` method writes an entire
+:class:`::arrow::Table` to an output file.
+
+The :class:`StreamReader` and :class:`StreamWriter` classes allow for
+data to be written using a C++ input/output streams approach to
+read/write fields column by column and row by row.  This is approach
+is offered for ease of use and type-safety.  It is of course also
+useful when data must be streamed as files are read and written
+incrementally.
```

Comment:
Should you add a word about performance? I presume it's not great compared to other IO facilities.